### DISCUSSION TITLE
Plotting values vs STL

### DISCUSSION THREAD

**Discussion by koenterheegde0507** [pyvista/pyvista#5794] [Q&A]
### First check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the PyVista documentation, with the integrated search.
- [X] I already searched in Google "How to X in PyVista" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I will NOT make duplicate discussions or issues in the future.

### Commit to Help

- [X] I commit to help with one of those options.

### Sample Code What is the problem, question, or error?
Write a short description telling me what you are doing, what you expect to happen, and what is currently happening.

```python
root = tk.Tk()
root.withdraw()
root.title('Pick patient data')
file_path = filedialog.askopenfilename()
chosenMesh = tm.load_mesh(f'{file_path}')
volume = tm.voxel.creation.voxelize(mesh=chosenMesh, pitch=chosenMesh.extents.max() / 200)
array = volume.matrix
filled = ndi.binary_fill_holes(array)

skeleton = skeletonize(filled)
x_skel, y_skel, z_skel = np.where(skeleton)
plotter = pv.Plotter()
coordinates = []
for x, y, z in zip(x_skel, y_skel, z_skel):
    coordinates.append((x,y,z))

coordinates = np.array(coordinates)   
plotter.add_points(coordinates,color='black')
plotter.add_mesh(chosenMesh,opacity = 0.5)
plotter.show()
```

### Description

Hello everyone, I'm currently using the 'skeletonize()' function from the scikit-image package and struggling with plotting the skeleton points at the same position as the mesh of the STL file. I first voxelize the mesh in order to apply the algorithm. I eventually get the coordinates of the points the skeleton is made out of, but they appear to be in a different position in comparison to the original mesh (see added pictures, also big difference between the cases in distance). Is this due to a plotting mistake or is this caused by the algorithm? Anyhow, I would like to have the points inside the mesh, but how could I apply a transformation to these points (I don't want to reposition the mesh). I thought of maybe using `mesh.center` but I don't know the center of the coordinates so I can't calculate the distance between those points. Also, the first/last coordinate do not start/end at the start/end of the mesh so those positions are also not common. Any ideas on how to solve this?

### System Information

```shell
Date: Mon Mar 18 21:47:17 2024 West-Europa (standaardtijd)

                OS : Windows
            CPU(s) : 8
           Machine : AMD64
      Architecture : 64bit
               RAM : 7.8 GiB
       Environment : Python
        GPU Vendor : Intel
      GPU Renderer : Intel(R) UHD Graphics 620
       GPU Version : 4.5.0 - Build 30.0.101.1692
  MathText Support : False

  Python 3.9.7 (tags/v3.9.7:1016ef3, Aug 30 2021, 20:19:38) [MSC v.1929 64 bit
  (AMD64)]

           pyvista : 0.42.3
               vtk : 9.3.0
             numpy : 1.24.3
        matplotlib : 3.8.2
            scooby : 0.9.2
             pooch : 1.8.0
            pillow : 10.1.0
           imageio : 2.33.0
             PyQt5 : 5.15.4
           IPython : 8.13.2
        ipywidgets : 8.1.2
             scipy : 1.9.1
              tqdm : 4.66.2
            meshio : 5.3.5
      nest_asyncio : 1.5.6
```

### Screenshots

![image](https://github.com/pyvista/pyvista/assets/152174547/1a1a7b2b-05e8-43f3-924a-f924cc1090a4)
![image](https://github.com/pyvista/pyvista/assets/152174547/b8c8d5f9-c57f-4c9d-9bb1-991a2467ca11)
![image](https://github.com/pyvista/pyvista/assets/152174547/6b9ef791-c341-4c64-a43b-98c011af4080)

**Comment by darikg** [comment]
just taking a stab in the dark here from glancing at your code but from

> x_skel, y_skel, z_skel = np.where(skeleton)

it looks like your skeleton coordinates are just voxel indices, i.e. if `nx, ny, nz = skeleton.shape` then `x_skel` is an integer in `[0, ..., nx - 1]`, etc. But your mesh has coordinates in some world space -- if you knew those limits `(x_world_min, x_world_max)`   then presumably you could rescale the skeleton voxel coordinates with something like `x_skel_world = x_world_min + (x_skel / (nx - 1)) * (x_world_max - x_world_min)` and so on
